neurons/validator/forward.py

```diff
@@ -140,37 +263,93 @@ def gen_caption(prompt, i):
         logger.error(f"Unable to log event to wandb due to the following error: {e}")
 
 
-async def run_step(
-    validator,
-    task: ImageGenerationTaskModel,
-    axons: List[AxonInfo],
-    uids: torch.LongTensor,
+async def create_batch_for_upload(
+    validator_wallet: bt.wallet,
+    metagraph: "bt.metagraph.Metagraph",
+    batch_id: str,
+    prompt: str,
+    responses: List[ImageGenerationResponse],
+    masked_rewards: MaskedRewards,
 ):
-    # Get Arguments
-    prompt = task.prompt
-    batch_id = task.task_id
-    task_type = task.task_type
+    uids = [response.uid for response in responses]
+
+    should_drop_entries = []
+    images = []
+    for response, reward in zip(responses, masked_rewards.rewards):
+        if response.has_images() and reward != 0:
+            im_file = BytesIO()
+            T.transforms.ToPILImage()(bt.Tensor.deserialize(response.images[0])).save(
+                im_file, format="PNG"
+            )
+            # im_bytes: image in binary format.
+            im_bytes = im_file.getvalue()
+            im_b64 = base64.b64encode(im_bytes)
+            images.append(im_b64.decode())
+            should_drop_entries.append(0)
+        else:
+            # Generated image has zero reward, we are dropping it
+            im_file = BytesIO()
+            T.transforms.ToPILImage()(
```

Comment:
Are we just saving a blank image here? We may want to keep blacklisted images (but not NSFW ones).